### PR TITLE
Add INX backend for fill analytics

### PR DESCRIPTION
**Pull request by Alex6323**
## Linked Issues

* Closes #

### CODE REVIEW

File: src/bin/inx-chronicle/cli/mod.rs [hunk 1]
```diff
@@ -335,8 +370,11 @@ pub enum Subcommands {
         /// Select a subset of analytics to compute.
         #[arg(long)]
         analytics: Vec<chronicle::db::influxdb::AnalyticsChoice>,
+        /// The input source to use for filling the analytics.
+        #[arg(long, value_name = "INPUT_SOURCE")]
+        input_source: InputSourceChoice,
```
Comment (Alexandcoats) [inline src/bin/inx-chronicle/cli/mod.rs:375]: Default this

File: src/bin/inx-chronicle/cli/analytics.rs [hunk 2]
```diff
@@ -7,42 +7,46 @@ use chronicle::{
...
-    db: &MongoDb,
-    influx_db: &InfluxDb,
+pub async fn fill_analytics<I: InputSource + Clone + Send + Sync + 'static>(
+    db: MongoDb,
+    influx_db: InfluxDb,
     start_milestone: MilestoneIndex,
     end_milestone: MilestoneIndex,
     num_tasks: usize,
-    analytics: &[AnalyticsChoice],
+    analytics: Vec<AnalyticsChoice>,
```
Comment (Alexandcoats) [inline src/bin/inx-chronicle/cli/analytics.rs:21]: But why

File: src/bin/inx-chronicle/cli/mod.rs [hunk 3]
```diff
@@ -2,6 +2,10 @@
 // SPDX-License-Identifier: Apache-2.0
 
 use chronicle::db::mongodb::config as mongodb;
+#[cfg(feature = "inx")]
```
Comment (Alexandcoats) [inline src/bin/inx-chronicle/cli/mod.rs:5]: Can we just use the full paths below

File: src/bin/inx-chronicle/cli/mod.rs (hunk 1 above)
Comment (Alex6323) [reply src/bin/inx-chronicle/cli/mod.rs:370]: Almost forgot this but luckily I didn't eventually. Thanks for reminding anyways.

File: src/bin/inx-chronicle/cli/analytics.rs (hunk 2 above)
Comment (Alex6323) [reply src/bin/inx-chronicle/cli/analytics.rs]: sorry ... I had some lifetime issues ... and this was an intermediary step.

File: src/bin/inx-chronicle/cli/mod.rs (hunk 3 above)
Comment (Alex6323) [reply src/bin/inx-chronicle/cli/mod.rs]: I felt exactly the same so I did this without even reading your comment. I think I have a small Alex Coats in my head.

File: src/tangle/sources/mod.rs [hunk 4]
```diff
@@ -47,7 +47,7 @@ pub struct BlockData {
 #[async_trait]
 pub trait InputSource
 where
-    Self: Send,
+    Self: Send + Sync,
```
Comment (Alexandcoats) [inline src/tangle/sources/mod.rs:50]: Could be moved above `InputSource: Send + Sync`

**Pull request merged**